src/github.com/couchbase/sync_gateway/rest/user_agent_version_test.go

```diff
@@ -0,0 +1,39 @@
+package rest
+
+import (
+	"testing"
+
+	"github.com/couchbaselabs/go.assert"
+)
+
+func TestUserAgentVersion(t *testing.T) {
+
+	userAgentVersion := NewUserAgentVersion("")
+	assert.Equals(t, userAgentVersion.MajorVersion(), 0)
+	assert.Equals(t, userAgentVersion.MinorVersion(), 0)
+
+	userAgentVersion = NewUserAgentVersion("CouchbaseLite/1.0.0")
+	assert.Equals(t, userAgentVersion.MajorVersion(), 1)
+	assert.Equals(t, userAgentVersion.MinorVersion(), 0)
+
+	userAgentVersion = NewUserAgentVersion("CouchbaseLite/1.1.0")
+	assert.Equals(t, userAgentVersion.MajorVersion(), 1)
+	assert.Equals(t, userAgentVersion.MinorVersion(), 1)
+
+	userAgentVersion = NewUserAgentVersion("CouchbaseLite/1.2 (suff/goes/here)")
+	assert.Equals(t, userAgentVersion.MajorVersion(), 1)
+	assert.Equals(t, userAgentVersion.MinorVersion(), 2)
+
+	userAgentVersion = NewUserAgentVersion("CouchbaseLite/1.2(suff.goes.here)")
+	assert.Equals(t, userAgentVersion.MajorVersion(), 1)
+	assert.Equals(t, userAgentVersion.MinorVersion(), 2)
+
+	userAgentVersion = NewUserAgentVersion("CouchbaseLite/1.2 suff.goes/here)")
+	assert.Equals(t, userAgentVersion.MajorVersion(), 1)
+	assert.Equals(t, userAgentVersion.MinorVersion(), 2)
+
+	userAgentVersion = NewUserAgentVersion("whatever/thing$$h3ysuz1!!///")
+	assert.Equals(t, userAgentVersion.MajorVersion(), 0)
+	assert.Equals(t, userAgentVersion.MinorVersion(), 0)
+
```

Comment:
I added a new unit test and updated the regex to look for a useragent starting with `CouchbaseLite` -- did a push -f with an amended commit.